app/services/course/duplication/course_duplication_service.rb

```diff
@@ -38,7 +38,10 @@ def duplicate_course(current_course, options = {}, all_objects = [], selected_ob
   # @return [Course] The duplicated course
   def duplicate_course(current_course)
     duplicator.duplicate(current_course).tap do |new_course|
-      notify_duplication_complete(new_course) if new_course.save
+      duplication_successful = Course.transaction do
```

Comment:
```
duplication_successful = Course.transaction do
  raise ActiveRecord::Rollback unless new_course.save
  raise AcriveRecord::Rollback unless update_course_settings(@duplicator, new_course, current_course)
  true
end
```

must raise a rollback so that the transaction will rollback when failure